docs/modules/ROOT/pages/how-tos/crossplane/working_with_functions.adoc

```diff
@@ -0,0 +1,197 @@
+    --set "xfn.args={--debug}"
+```
+
+
+== Preparation step 2
+Next step is to declare our Function in our managed resource (Composition).
+It's highly recommended to use other container registry than hub.docker.io especially for development because their pull limit can block Your dev environment
+```
+spec:
+  compositeTypeRef:
+    apiVersion: vshn.appcat.vshn.io/v1
+    kind: XVSHNPostgreSQL
+  functions:
+    - name: fnio
+      type: Container
+      container:
+        image: ghcr.io/wejdross/fnio
+        imagePullPolicy: Always
+```
+
+
+== Example function 1 - Golang
+
+As a result of code below there will be ConfigMap created even though it wasn't declared in original composition. This code simply takes input from stdin and unmarshal it into xfnv1alpha1.FunctionIO{}, further we append current list of desired resources by our new resource and print it to stdout. You can see that we also create log entry which will print whole aboject we received - be carefull with that, as in my case it was almost 5000 lines. 
+
+```
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	xfnv1alpha1 "github.com/crossplane/crossplane/apis/apiextensions/fn/io/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+func main() {
+
+	funcIO := xfnv1alpha1.FunctionIO{}
+
+	x, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		log.Fatal(err)
+	}
+    // it's necessary to use sigs.k8s.io/yaml parser, otherwise it's impossible to force functions to work
+	err = yaml.Unmarshal(x, &funcIO)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	object := `
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: Object
+metadata:
+  name: testingconfigmaps
+spec:
+  providerConfigRef:
+    name: kubernetes
+  forProvider:
+    manifest:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: url-config
+        namespace: default
+      data:
+        fullURL: "https://google.pl"`
+
+	k8sapproved, err := yaml.YAMLToJSON([]byte(object))
+	if err != nil {
+		log.Fatal("from k8sapproved", err)
+	}
+
+	funcIO.Desired.Composite.Resource.Raw = funcIO.Observed.Composite.Resource.Raw
+
+	funcIO.Desired.Resources = append(funcIO.Desired.Resources, xfnv1alpha1.DesiredResource{
+		Name: "examplename",
+		Resource: runtime.RawExtension{
+			Raw: k8sapproved,
+		},
+	},
+	)
+
+	funcIO.Results = append(funcIO.Results,
+		xfnv1alpha1.Result{
+			Severity: xfnv1alpha1.SeverityNormal,
+			Message:  fmt.Sprintf("Print whole object: \n\n\n%s\n\n\n", string(x)),
+		},
+	)
+
+	d1, err := yaml.Marshal(funcIO)
+	if err != nil {
+		log.Fatal(err)
+	}
+    // return new state to stdout
+	fmt.Println(string(d1))
+}
+```
+
+Example Dockerfile to build container:
+
+```
+FROM golang:latest AS build
+WORKDIR /build
+COPY go.mod /build/
+COPY go.sum /build/
+RUN go mod download
+COPY main.go /build/
+RUN go build -o /build/fnio
+
+FROM ubuntu:latest
+COPY --from=build /build/fnio /fnio
+ENTRYPOINT [ "/fnio" ]
+```
+
+== Example 2 - Python
```

Comment:
Can you add the initial composition here as well like you did for go example? It should be much clearer what is being updated.